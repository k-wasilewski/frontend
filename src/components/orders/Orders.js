import React, {Component} from 'react';
import '../../css/App.css';
import ItemList from "./ItemList";
import NewOrderStep2 from "./NewOrderStep2";

class Orders extends Component {

    render() {
        return (
            <div>
                <NewOrderStep2 />
                <ItemList />
            </div>
        );
    };
}

export default Orders;
