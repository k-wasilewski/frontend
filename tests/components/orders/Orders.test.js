import React from 'react';
import renderer from 'react-test-renderer';
import store from "../../../src/redux/store";
import Orders from "../../../src/components/orders/Orders";
import {Provider} from "react-redux";

describe("Orders specification", () => {
    it('Orders renders NewOrderStep2 and ItemList components', () => {
        const component = renderer.create(
            <Provider store={store}>
                <Orders />
            </Provider>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        expect((/Kolor:/g).test(JSON.stringify(tree))).toBeTruthy()
        expect((/Rozmiar:/g).test(JSON.stringify(tree))).toBeTruthy()
        expect((/Lista zamówień/g).test(JSON.stringify(tree))).toBeTruthy()
    })
})