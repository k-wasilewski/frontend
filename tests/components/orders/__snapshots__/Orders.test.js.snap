// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders rendering specification renders NewOrderStep2 and ItemList components 1`] = `
<div>
  <div
    className="main"
  >
    <form
      onSubmit={[Function]}
    >
      <p
        className="resp"
      >
        
      </p>
      <div
        className="form"
      >
        <div
          className="col1"
        >
           
          ,
           
          <p>
            <label
              htmlFor="color"
            >
              Kolor:
            </label>
          </p>
          <p>
            Rozmiar:
          </p>
          <button>
            Zapisz
          </button>
          <button
            onClick={[Function]}
          >
            Wyślij
          </button>
        </div>
        <div
          className="col2"
        >
          <p>
            <select
              id="color"
              name="color"
              onChange={[Function]}
              value=""
            >
              <option
                defaultValue={true}
                disabled={true}
                value=""
              >
                (wybierz)
              </option>
              <option
                value="blue"
              >
                Niebieski
              </option>
              <option
                value="lightblue"
              >
                Błękitny
              </option>
              <option
                value="darkblue"
              >
                Granatowy
              </option>
            </select>
          </p>
          <input
            checked={false}
            id="s"
            name="size"
            onChange={[Function]}
            type="radio"
            value="s"
          />
          <label
            htmlFor="male"
          >
            S
          </label>
          <br />
          <input
            checked={false}
            id="m"
            name="size"
            onChange={[Function]}
            type="radio"
            value="m"
          />
          <label
            htmlFor="female"
          >
            M
          </label>
          <br />
          <input
            checked={false}
            id="l"
            name="size"
            onChange={[Function]}
            type="radio"
            value="l"
          />
          <label
            htmlFor="other"
          >
            L
          </label>
          <br />
          <input
            checked={false}
            id="xl"
            name="size"
            onChange={[Function]}
            type="radio"
            value="xl"
          />
          <label
            htmlFor="other"
          >
            XL
          </label>
          <br />
        </div>
        <div
          className="col2"
        >
          <div
            id="visualization"
          >
             
          </div>
        </div>
      </div>
    </form>
  </div>
  <div
    className="main"
  >
    <h3>
      Lista zamówień
    </h3>
    <ol
      start="1"
    />
  </div>
</div>
`;

exports[`Orders specification renders NewOrderStep2 and ItemList components 1`] = `
<div>
  <div
    className="main"
  >
    <form
      onSubmit={[Function]}
    >
      <p
        className="resp"
      >
        
      </p>
      <div
        className="form"
      >
        <div
          className="col1"
        >
           
          ,
           
          <p>
            <label
              htmlFor="color"
            >
              Kolor:
            </label>
          </p>
          <p>
            Rozmiar:
          </p>
          <button>
            Zapisz
          </button>
          <button
            onClick={[Function]}
          >
            Wyślij
          </button>
        </div>
        <div
          className="col2"
        >
          <p>
            <select
              id="color"
              name="color"
              onChange={[Function]}
              value=""
            >
              <option
                defaultValue={true}
                disabled={true}
                value=""
              >
                (wybierz)
              </option>
              <option
                value="blue"
              >
                Niebieski
              </option>
              <option
                value="lightblue"
              >
                Błękitny
              </option>
              <option
                value="darkblue"
              >
                Granatowy
              </option>
            </select>
          </p>
          <input
            checked={false}
            id="s"
            name="size"
            onChange={[Function]}
            type="radio"
            value="s"
          />
          <label
            htmlFor="male"
          >
            S
          </label>
          <br />
          <input
            checked={false}
            id="m"
            name="size"
            onChange={[Function]}
            type="radio"
            value="m"
          />
          <label
            htmlFor="female"
          >
            M
          </label>
          <br />
          <input
            checked={false}
            id="l"
            name="size"
            onChange={[Function]}
            type="radio"
            value="l"
          />
          <label
            htmlFor="other"
          >
            L
          </label>
          <br />
          <input
            checked={false}
            id="xl"
            name="size"
            onChange={[Function]}
            type="radio"
            value="xl"
          />
          <label
            htmlFor="other"
          >
            XL
          </label>
          <br />
        </div>
        <div
          className="col2"
        >
          <div
            id="visualization"
          >
             
          </div>
        </div>
      </div>
    </form>
  </div>
  <div
    className="main"
  >
    <h3>
      Lista zamówień
    </h3>
    <ol
      start="1"
    />
  </div>
</div>
`;
